sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Catalog.scala

```diff
@@ -81,7 +84,7 @@ trait Catalog {
 }
 
 class SimpleCatalog(val conf: CatalystConf) extends Catalog {
-  val tables = new mutable.HashMap[String, LogicalPlan]()
+  val tables = new ConcurrentHashMap[String, LogicalPlan] asScala
```

Comment:
Actually it's a bad idea to put a scala wrapper around java's ConcurrentHashMap. Do you mind just removing the wrapper, and change the call site to just use the functions defined in ConcurrentHashMap?

Also would be great to mark tables as a private val.